docs/how-to-upgrade.md

```diff
@@ -0,0 +1,53 @@
+# Upgrade notes
+
+This document highlights the required actions for upgrading to the latest release.
+
+## Upgrading to Key Vault provider 0.0.9+
+
+**tl;dr** - :warning: `0.0.9+` release of the Azure Key Vault provider is incompatible with the Secrets Store CSI Driver versions < `v0.0.14`.
```

Comment:
nit: should we mention driver 0.0.9+ still works with older version of the provider?

Reply:
@ritazh Do you mean driver `v0.0.14` still works with releases of provider prior to `0.0.9`?

Reply:
Yes 🤦‍♀️